db/db.go

```diff
@@ -169,6 +222,29 @@ func (db *Database) InsertState(path string, versionID string, sf *statefile.Fil
 	return nil
 }
 
+// UpdateState update a Terraform State in the Database with Lineage foreign constraint
+// It will also insert Lineage entry in the db if needed
+func (db *Database) UpdateState(st types.State) error {
+	// Get lineage from old column
+	if err := db.Raw("SELECT lineage FROM states WHERE path = ?", st.Path).Scan(&st.Lineage.Value).Error; err != nil {
```

Comment:
Isn't there a way do do that using the Go structs?

Reply:
Not to my knowledge, knowing that it is a request to recover an element which is no longer present in the structure during the migration.
There is however the ```MigrateColomn``` method of Gorm's Migrator interface (https://gorm.io/docs/migration.html) but I'm not sure what it does, I didn't understand the source code and there is no documentation...